### PR TITLE
fix: Make resource not exists regex less agressive

### PR DESCRIPTION
**Pull request by erezrokah**
🎉 Thank you for making CloudQuery awesome by submitting a PR 🎉

#### Summary

Follow up from https://github.com/cloudquery/cq-provider-aws/pull/821#discussion_r863583321

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Update or add tests. Learn more about testing [here](https://docs.cloudquery.io/docs/developers/sdk/testing) 🧪
- [x] Update the docs by running `go run ./docs/docs.go` and committing the changes 📃
- [x] If adding a new resource, add [relevant Terraform files](../terraform) in a separate PR 📂
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

File: client/errors.go [hunk 1]
```diff
@@ -157,16 +157,16 @@ func isCodeThrottle(code string) bool {
...
+	hostIdRegex            = regexp.MustCompile(`\sHostID: [A-Za-z0-9+/_=-]+`)
+	arnIdRegex             = regexp.MustCompile(`(\s)(arn:aws[A-Za-z0-9-]*:)[^ \.\(\)\[\]\{\}\;\,]+(\s?)`)
+	urlRegex               = regexp.MustCompile(`([\s"])http(s?):\/\/[a-z0-9_\-\./]+([":\s]?)`)
+	lookupRegex            = regexp.MustCompile(`(\slookup\s)[-A-Za-z0-9\.]+\son\s([0-9]{0,3}\.[0-9]{0,3}\.[0-9]{0,3}\.[0-9]{0,3}:[0-9]{1,5})(:.+?)([0-9]{0,3}\.[0-9]{0,3}\.[0-9]{0,3}\.[0-9]{0,3}:[0-9]{1,5})->([0-9]{0,3}\.[0-9]{0,3}\.[0-9]{0,3}\.[0-9]{0,3}:[0-9]{1,5})(:.*)`)
+	dialRegex              = regexp.MustCompile(`(\sdial\s)(tcp|udp)(\s)([0-9]{0,3}\.[0-9]{0,3}\.[0-9]{0,3}\.[0-9]{0,3}:[0-9]{1,5})(:.+?)`)
+	encAuthRegex           = regexp.MustCompile(`(\s)(Encoded authorization failure message:)\s[A-Za-z0-9_-]+`)
+	userRegex              = regexp.MustCompile(`(\s)(is not authorized to perform: .+ on resource:\s)(user)\s.+`)
+	s3Regex                = regexp.MustCompile(`(\s)(S3(Key|Bucket))=(.+?)([,;\s])`)
+	resourceNotExistsRegex = regexp.MustCompile(`(\sThe )([A-Za-z0-9 -]+ )'([A-Za-z0-9 -]+?)'( does not exist)`)
```
Comment (disq) [inline client/errors.go:169]: $3 could use more chars? underscore, slash? In fact looks like the format for resource IDs is documented: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/resource-ids.html

Comment (erezrokah) [reply client/errors.go]: Thanks, done in https://github.com/cloudquery/cq-provider-aws/pull/827/commits/8de76dae5456fd822ad830e11c25aa01229f64f8 plus an additional error

File: client/errors_test.go [hunk 2]
```diff
@@ -51,6 +51,10 @@ func TestRemovePII(t *testing.T) {
 			"operation error EC2: DescribeSnapshotAttribute, https response error StatusCode: 400, RequestID: xxxx, api error InvalidSnapshot.NotFound: The snapshot 'snap-11111111111111111' does not exist.",
 			"operation error EC2: DescribeSnapshotAttribute, https response error StatusCode: 400, RequestID: xxxx, api error InvalidSnapshot.NotFound: The snapshot 'xxxx' does not exist.",
 		},
+		{
+			"AutoScalingGroup name not found - Could not find Auto Scaling group named 'hub-dev-1111'",
```
Comment (disq) [inline client/errors_test.go:55]: Name leak - shouldn't include real-life names in tests

Comment (erezrokah) [reply client/errors_test.go]: Fixed! thanks :)

**Review by disq — APPROVED** [review]

**Pull request merged**